packages/core/util/layouts/GranularRectLayout.ts

```diff
@@ -139,7 +139,9 @@ class LayoutRow<T> {
         )
         this.initialize(left, right)
       } else if (additionalLength > 0) {
-        this.rowState.bits.push(...new Array(additionalLength))
```

Comment:
What exactly do you mean by the spread was "bad"? Was is just that the spread was too big to handle?

Reply:
the spread literally pushes all the arguments onto the stack, whereas a concat doesn't. I just used the word bad to mean that this can result in bad behavior e.g. maximum stack size exceeded and is often an unintended result of using spread